src/NexusMods.App.UI/Pages/Downloads/DownloadsPageViewModel.cs

```diff
@@ -11,28 +18,211 @@ namespace NexusMods.App.UI.Pages.Downloads;
+                .Delay(TimeSpan.FromMilliseconds(64))
+                .OnUI()
+                .Select(_ => Unit.Default);
+            
+            // Create observables that track selection and status changes
+            var selectionStatusReevaluationTrigger = this.WhenAnyValue(vm => vm.SelectionCount)
+                .Select(_ => Unit.Default)
+                .Merge(runningCountChanges);
+
+            SelectionHasRunningItems = selectionStatusReevaluationTrigger
+                .Select(_ => Adapter.SelectedModels
+                    .Select(model => model.GetOptional<DownloadComponents.StatusComponent>(DownloadColumns.Status.ComponentKey))
+                    .Where(opt => opt.HasValue)
+                    .Any(opt => opt.Value.Status.Value == JobStatus.Running))
+                .ToObservable();
+            
+            SelectionHasPausedItems = selectionStatusReevaluationTrigger
+                .Select(_ => Adapter.SelectedModels
+                    .Select(model => model.GetOptional<DownloadComponents.StatusComponent>(DownloadColumns.Status.ComponentKey))
+                    .Where(opt => opt.HasValue)
+                    .Any(opt => opt.Value.Status.Value == JobStatus.Paused))
+                .ToObservable();
+            
+            SelectionHasActiveItems = selectionStatusReevaluationTrigger
+                .Select(_ => Adapter.SelectedModels
+                    .Select(model => model.GetOptional<DownloadComponents.StatusComponent>(DownloadColumns.Status.ComponentKey))
+                    .Where(opt => opt.HasValue)
+                    .Any(opt => opt.Value.Status.Value.IsActive()))
+                .ToObservable();
+            
+            // Create context-aware observables for running and paused items
+            HasRunningItems = context.GameScope.HasValue
+                ? downloadsService.GetRunningDownloadsForGame(context.GameScope.Value)
+                    .QueryWhenChanged(items => items.Count > 0)
+                    .OnUI()
+                    .ToObservable()
+                    .Prepend(false)
+                : downloadsService.GetDownloadsByStatus(JobStatus.Running)
+                    .QueryWhenChanged(items => items.Count > 0)
+                    .OnUI()
+                    .ToObservable()
+                    .Prepend(false);
+            
+            HasPausedItems = context.GameScope.HasValue
+                ? downloadsService.GetPausedDownloadsForGame(context.GameScope.Value)
+                    .QueryWhenChanged(items => items.Count > 0)
+                    .OnUI()
+                    .ToObservable()
+                    .Prepend(false)
+                : downloadsService.GetDownloadsByStatus(JobStatus.Paused)
+                    .QueryWhenChanged(items => items.Count > 0)
+                    .OnUI()
+                    .ToObservable()
+                    .Prepend(false);
```

Comment:
I would keep these outside the `WhenActivated` so that we don't recreate the observable theselves.
 
We just need to do the subscribe part in the `WhenActivated`, the rest we can prepare in the constructor.

Reply:
Not sure if we have a way to re-activate in our codebase, because navigating away the page does a full teardown.
And we don't have a thing where we suddenly show/hide.

I did move them nonetheless in e9554572351d7e7d2bc8855a7d5294e50c9d8773 for the purpose of correctness; though ultimately, there's little difference in terms of runtime characteristics. Once everything's unsubscribed on deactivation, the pipeline ceases to exist. So it's just a matter of nanoseconds on activation.